src/gui/DatabaseOpenDialog.cpp

```diff
@@ -84,9 +84,8 @@ void DatabaseOpenDialog::showEvent(QShowEvent* event)
 {
     QDialog::showEvent(event);
     QTimer::singleShot(100, this, [this] {
-        if (m_view->isOnQuickUnlockScreen() && !m_view->unlockingDatabase()) {
-            m_view->triggerQuickUnlock();
-        }
+        // Automatically trigger quick unlock if it's available
+        m_view->triggerQuickUnlock();
```

Comment:
I'm 60% sure that there was a reason why we had this extra check for isOnQuickUnlockScreen().

Reply:
The checks are moved into trigger quick unlock

Reply:
I checked that and I could have sworn the isOnQuickUnlockScreen() check was missing, but apparently I'm mistaken.